rioting-daisies/BangazonAPI
# Implementation of CRUD for the department resource

# Description
Created all GET, PUT, POST, and DELETE methods for the department controller as well as all tests associated with those methods.


Fixes Issue # https://github.com/rioting-daisies/BangazonAPI/issues/6
## Type of change

Please delete options that are not relevant.




# Testing Instructions
Git fetch --all
checkout into the cm-product-controller branch
Make sure your data for departments matches (Department 1, HR, 25000)
Review code for any errors
Run test in TestDepartment and ensure everything works correctly.


# Checklist:

- [x ] My code follows the style guidelines of this project
- [x ] I have performed a self-review of my own code
- [ x] I have commented my code, particularly in hard-to-understand areas
- [ x] I have made corresponding changes to the documentation
- [x ] My changes generate no new warnings
- [ x] I have added test instructions that prove my fix is effective or that my feature works
